BuildingRefrigerator/BuildingRefrigerator/refrigerator.cs

```diff
@@ -0,0 +1,140 @@
+﻿namespace BuildingRefrigerator;
+
+public class refrigerator : IComparable<refrigerator?>
+{
+    static private int _uniqueIdentifierRefrigerator = 0;
+    private List<shelf?>? _shelves;
+    private string? _model;
+    private int _maximumShelves;
+    public refrigerator(string? model, string color, int maximumShelves, List<shelf?>? shelves = null)
+    {
+        Id = ++_uniqueIdentifierRefrigerator;
+        Model = model;
+        Color = color;
+        _maximumShelves = maximumShelves;
+        Shelves = shelves;
+    }
+    public int Id { get; }
+    public string? Model
+    {
+        get=> _model;
+        set
+        {
+            if (value==null)
+                throw new Exception("A refrigerator model is missing. Cannot create a fridge");
+            _model = value;
+        }
+    }
+    public string Color { get; set; }
+    public List<shelf?>? Shelves
+    {
+        get => _shelves;
+        set
+        {
+            if (value is not null)
+            {
+                _shelves = value;
+            }
+            else
+            {
+                _shelves = new List<shelf?>();
+                for (int i = 1; i <= _maximumShelves; i++)
+                    _shelves.Add(new shelf(i));
+            }
+        }
+    }
+    public override string ToString()
+    {
+        return "refrigerator:  "+ this.ToStringProperty();
+    }
+    public double HowMuchSpaceIsLeftOnTheRefrigerator()
+    {
+        return this.Shelves.Sum(shelf => shelf?.HowMuchSpaceIsLeftOnTheShelf() ?? throw new Exception("Error. No shelf found"));
+    }
+    public void InsertingAnItem(Item myItem)
+    {
+        shelf? myShelf = this?.Shelves?.FirstOrDefault(shelf => shelf?.HowMuchSpaceIsLeftOnTheShelf() >= myItem.Place);
+        if (myShelf == null)
+            throw new Exception("there are no place in the refrigerator");
+        else
+        {
+            myItem.IdShelf = myShelf.Id;
+            myShelf?.Items?.Add(myItem);
```

Comment:
Why are you able to access the items collection directly? should be wrapped in method.